### PR TITLE
VEN-1221 | Redirect the user to login after step 2

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
## Description :sparkles:
- Make all form tabs private routes
- Refactor the selection page to use the same layout for both berth and winter storage applications
- Remove the switch application fields from the selection page

### Closes :no_good_woman:
[**VEN-1221**](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1221): Redirect the user to the login view in all applications

### Related :handshake:
[VEN-1404](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1404): Transfer application: Change the current berth view to the customer info view

### Manual testing :construction_worker_man:
- Go to the second step (selected harbors or winter storage areas) without logging in
- The selection page no longer shows a note whether this is a new application or a transfer application
- On the same page you would see a note about the authentication
- The next button redirects the user to the login page
- After logging in, the user should be back to the third step of the form (with the same language option selected)
- Trying to access any of the routes manually (e.g. by typing the route to the third step in the browser) without logging in, should redirect the user to the login page.
- If the user repeats the same process but with logging in first, the second step doesn't show the authentication note, and the next button leads directly to the third step.

## Screenshots :camera_flash:
image

## Additional notes :spiral_notepad:
- Switch application fields to be added to the application info view in another PR

### CODE REVIEW

File: src/features/winterStorage/selectedAreaPage/SelectedAreaPageContainer.tsx [hunk 1]
```diff
@@ -75,21 +90,59 @@ const UnconnectedSelectedAreaPage = ({ localePush, values, selectedAreas, select
   const selected = getSelectedResources(selectedAreas, areas);
   const validSelection = selected.every(filter);
 
+  const handleSubmitApplication = (values: WinterFormValues) => {
+    // if (submitSwitchForm) {
+    //   submitSwitchForm(values);
+    // }
```
Comment (Jokinen) [inline src/features/winterStorage/selectedAreaPage/SelectedAreaPageContainer.tsx:96]: Commented out code.

**Review by Jokinen — COMMENTED** [review]
Code looks good to me. I can't quite understand the way in which `tContext` is used--I don't understand why it's necessary. But I guess that it's a choice that was made before this PR so it isn't that relevant here.

File: src/common/selectionPageLayout/SelectionPageLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import './selectionPage.scss';
+import { BoatInfo } from '../../features/berth/selectedBerthPage/types';
+import { TContext } from '../types/translation';
+
+export type Props = {
+  tContext: TContext;
+  boatInfo?: BoatInfo;
+  legend: { title: string; legend: string };
+  submitDisabled: boolean;
+  SelectionElements: List<JSX.Element>;
```
[the remark this answers is not in the document]
Comment (Nizar-Rahme) [reply src/common/selectionPageLayout/SelectionPageLayout.tsx:20]: Yes, that's correct

File: src/features/winterStorage/selectedAreaPage/SelectedAreaPageContainer.tsx (hunk 1 above)
Comment (Nizar-Rahme) [reply src/features/winterStorage/selectedAreaPage/SelectedAreaPageContainer.tsx:96]: Good catch. I will remove it.

File: src/common/selectionPageLayout/SelectionPageLayout.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jokinen) [reply src/common/selectionPageLayout/SelectionPageLayout.tsx:20]: So shouldn't it be lowecased in that case?

**Comment by Jokinen** [comment]
I didn't approve because of the one broken tests, otherwise this looked good to me.

**Comment by Nizar-Rahme** [comment]
> Code looks good to me. I can't quite understand the way in which `tContext` is used--I don't understand why it's necessary. But I guess that it's a choice that was made before this PR so it isn't that relevant here.

Most of the components are the same for both "berth places" and "winter storage areas". The differences are mainly in the text. So instead of passing every single string as a prop to the component, I decided to take advantage of [the context feature in i18n](https://www.i18next.com/translation-function/context) and let the translation files determine the corresponding text.
In addition to having fewer props, translation strings can live next to each other like the example below, so adjusting one string can be a reminder to adjust the other :):
```JSON
{
  "xxx_berth": "select berth",
  "xxx_winter": "select area"
}
```

Other suggestions and improvements are welcome. This was always an issue for this project, we should either duplicate a lot of components or pass tons of props. To me, this looks like a fine balance.

**Comment by Jokinen** [comment]
@Nizar-Rahme Thank you for the explanation, I get it now.

I don't oppose this approach and I see the benefits. My only concern is whether it'll prove to be a little brittle or hard to reason about when teams and devs change. In those cases the verbose but very straightforward option can sometimes be easier to maintain. But that's not reason enough to change the approach.

File: src/features/winterStorage/selectedAreaPage/SelectedAreaPageContainer.tsx (hunk 1 above)
Comment (Nizar-Rahme) [reply src/features/winterStorage/selectedAreaPage/SelectedAreaPageContainer.tsx]: Done

File: src/common/selectionPageLayout/SelectionPageLayout.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nizar-Rahme) [reply src/common/selectionPageLayout/SelectionPageLayout.tsx]: Yup :)
Fixed

**Review by Jokinen — APPROVED** [review]

**Comment by Nizar-Rahme** [comment]
Merging, I believe browser tests are not working against the review environment due to missing configurations, but they should work against staging 🤞 
If not, I will work on a follow-up fix.

**Pull request merged**